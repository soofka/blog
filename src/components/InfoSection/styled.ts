import styledComponents from 'styled-components';

export const StyledInfoSection = styledComponents.div`
  position: static;
  display: inline-block;
  width: 100%;
  height: ${props => props.theme.infoSectionHeightValue}px;
  color: ${props => props.theme.darkerWhite};
  background-color: ${props => props.theme.blue};
  box-shadow: ${props => `0 0 ${props.theme.baseDistanceValue}px 0 ${props.theme.darkerBlue}`};

  a {
    color: ${props => props.theme.darkerWhite};
    text-decoration: none;

    &:hover,
    &:active {
      color: ${props => props.theme.white};
    }
  }
`;

// export const StyledInfoSection = styledComponents.div`
//   position: absolute;
//   top: 0;
//   left: -${props => props.theme.infoSectionWidthValue}px;
//   box-sizing: border-box;
//   width: ${props => props.theme.infoSectionWidthValue}px;
//   height: auto;
//   min-height: 100%;
//   padding: ${props => props.theme.paddingS};
//   color: ${props => props.theme.darkerWhite};
//   background-color: ${props => props.theme.blue};
//   box-shadow: ${props => `0 0 ${props.theme.baseDistanceValue}px 0 ${props.theme.darkerBlue}`};
//
//   a {
//     color: ${props => props.theme.darkerWhite};
//     text-decoration: none;
//
//     &:hover,
//     &:active {
//       color: ${props => props.theme.white};
//     }
//   }
//
//   @media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
//     position: static;
//     display: inline-block;
//     width: 100%;
//     height: ${props => props.theme.infoSectionMobileHeightValue}px;
//     min-height: 0;
// }
// `;

export default StyledInfoSection;

