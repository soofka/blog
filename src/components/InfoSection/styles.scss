@import 'components/Blog/variables.scss';

.info-section {
  position: fixed;
  top: 0;
  bottom: 0;
  box-sizing: border-box;
  width: $infoSectionWidth;
  height: auto;
  min-height: 100%;
  padding: $paddingS;
  color: $darkerWhite;
  background-color: $blue;
  box-shadow: 0 0 $baseDistance 0 $darkerBlue;

  a {
    color: $darkerWhite;
    text-decoration: none;

    &:hover,
    &:active {
      color: $white;
    }
  }
}

@media (max-width: $mobileBreakPointWidth) {
  .info-section {
    position: static;
    display: inline-block;
    width: 100%;
    min-height: 0;
  }
}
