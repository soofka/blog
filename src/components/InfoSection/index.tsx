import * as React from 'react';

import Logo from 'components/Logo';
import Quote from 'components/Quote';
import LinkIconsBar from 'components/LinkIconsBar';
import LanguageSwitcher from 'components/LanguageSwitcher';

import './styles.scss';

export const InfoSection = (): JSX.Element => {
  return (
    <div className="info-section">
      <Logo/>
      {/*<LanguageSwitcher/>*/}
      <Quote/>
      <LinkIconsBar/>
    </div>
  );
};

export default InfoSection;
