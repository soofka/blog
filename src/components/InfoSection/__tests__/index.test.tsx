import * as React from 'react';
import { shallow } from 'enzyme';

import InfoSection from 'components/InfoSection';

describe('<InfoSection/>', () => {

  it('renders correctly', () => {
    expect(shallow(<InfoSection/>)).toMatchSnapshot();
  });

});
