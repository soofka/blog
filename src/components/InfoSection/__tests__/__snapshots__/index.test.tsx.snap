exports[`<InfoSection/> renders correctly 1`] = `
<div
  className="info-section">
  <Component />
  <Component />
  <Component />
</div>
`;
